W3ChampionsStatisticService/Ladder/RankRepository.cs

```diff
@@ -118,6 +147,22 @@ private async Task<List<Rank>> JoinWith(Expression<Func<Rank,bool>> matchExpress
             return result.Where(r => r.Player != null).ToList();
         }
 
+        private async Task<List<Rank>> JoinWithSortedByLeague(Expression<Func<Rank,bool>> matchExpression)
+        {
+            var ranks = CreateCollection<Rank>();
+            var players = CreateCollection<PlayerOverview>();
+            var result = await ranks
+                .Aggregate()
+                .Match(matchExpression)
```

Comment:
Did you have any problems with `Match` part of this pipeline failing because it cant parse `.contains`?

I recommend writing an integration test with the DB for this.